plugins/Ntuplizer.cc

```diff
@@ -522,211 +553,247 @@ Ntuplizer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+	      LeptonWeightUnc  += pow(theMuonAnalyzer->GetMuonTriggerSFErrorIsoMu24(MuonVect.at(m1)),2);
+	    }
+	    else {
+	      LeptonWeight     *= theMuonAnalyzer->GetMuonTriggerSFIsoMu24(TightMuonVect.at(m2));
+	      LeptonWeightUnc  += pow(theMuonAnalyzer->GetMuonTriggerSFErrorIsoMu24(MuonVect.at(m2)),2);
+	    }
+	  }// IsoMu24 trigger
+	  if (isMu50_OR_TkMu50) {
+	    if (TightMuonVect.at(m1).pt() > TightMuonVect.at(m2).pt() ) {
+	      LeptonWeight     *= theMuonAnalyzer->GetMuonTriggerSFMu50(TightMuonVect.at(m1));
+	      LeptonWeightUnc  += pow(theMuonAnalyzer->GetMuonTriggerSFErrorMu50(MuonVect.at(m1)),2);
+	    }
+	    else {
+	      LeptonWeight     *= theMuonAnalyzer->GetMuonTriggerSFMu50(TightMuonVect.at(m2));
+	      LeptonWeightUnc  += pow(theMuonAnalyzer->GetMuonTriggerSFErrorMu50(MuonVect.at(m2)),2);
+	    }
+	  }// IsoMu50 trigger
 
+	  // //removed obsolete things for now
+	  // LeptonWeight *= theMuonAnalyzer->GetMuonTrkSF(TightMuonVect.at(m1));
+	  // LeptonWeight *= theMuonAnalyzer->GetMuonTrkSF(TightMuonVect.at(m2));
+    	  // //LeptonWeightUnc += pow(theMuonAnalyzer->GetMuonTrkSFError(MuonVect.at(m1))      ,2);
+    	  // //LeptonWeightUnc += pow(theMuonAnalyzer->GetMuonTrkSFError(MuonVect.at(m2))      ,2);
 
-    //}
-    //}
+    	  LeptonWeight *= theMuonAnalyzer->GetMuonIdSF(TightMuonVect.at(m1), 1);
+    	  LeptonWeight *= theMuonAnalyzer->GetMuonIdSF(TightMuonVect.at(m2), 1);
+    	  LeptonWeight *= theMuonAnalyzer->GetMuonIsoSF(TightMuonVect.at(m1), 1);
+    	  LeptonWeight *= theMuonAnalyzer->GetMuonIsoSF(TightMuonVect.at(m2), 1);
 
-    //else if(isZtoEE) {
-    //   if(isVerbose) std::cout << "Do the Z->e e" << std::endl;
-    //  int e1(-1), e2(-1);
-    //      for(unsigned int i = 0; i < TightElecVect.size(); i++) {
-    //	for(unsigned int j = 1; j < TightElecVect.size(); j++) {
-    //	  if(i==j || TightElecVect[i].charge() == TightElecVect[j].charge())
-    //	    {
-    //	      isZtoEE = false;
-    //	      continue;
-    //	    }
-    //	  float Zmass = (TightElecVect[i].p4() + TightElecVect[j].p4()).mass();
-    //	  if(Zmass > 50. && Zmass < 130.)
-    //	    {
-    //	      e1 = i;
-    //	      e2 = j;
-    //	      isZtoEE =true;
-    //	    }
-    //	}
-    //}
-    //// Build candidate
-    //if(e1 >= 0 && e2 >= 0) {
-    //	theZ.addDaughter(TightElecVect.at(e1).charge() ? TightElecVect.at(e1) : TightElecVect.at(e2));
-    //	theZ.addDaughter(TightElecVect.at(e1).charge() ? TightElecVect.at(e2) : TightElecVect.at(e1));
-    //	addP4.set(theZ);
-    //	isZtoEE = true;
-    //
+    	  LeptonWeightUnc += pow(theMuonAnalyzer->GetMuonIdSFError(MuonVect.at(m1), 1)    ,2);
+    	  LeptonWeightUnc += pow(theMuonAnalyzer->GetMuonIdSFError(MuonVect.at(m2), 1)    ,2);
+    	  LeptonWeightUnc += pow(theMuonAnalyzer->GetMuonIsoSFError(MuonVect.at(m1), 1)   ,2);
+    	  LeptonWeightUnc += pow(theMuonAnalyzer->GetMuonIsoSFError(MuonVect.at(m2), 1)   ,2);
```

Comment:
SF computed for `TightMuonVect` and SF errors for `MuonVect`. Is this intended?

Reply:
It looks like a mis-naming of the vector. @MelanieEich ?

Reply:
In that case, one should also check the `id` used

Reply:
Please use here TightMuonVect.at(m2) and id=3.

Reply:
Done!